getting-to-know-angularjs.md

```diff
@@ -0,0 +1,197 @@
+
+**_Note! If you try to validate this code with https://validator.w3.org/#validate_by_input you'll get some validation errors. 
+You can use `data-ng-` instead of `ng-` if you want to make your page HTML valid._**
+
+### Modules and Controllers
+
+So far so good, but there's not much of an application here. Let's introduce modules and controllers and see what they give us.
+
+In AngularJS **modules** define applications while **controllers** control the data and logic. Controllers always belong to a module.
+
+You can add modules and controllers using the `<script>` tag or store them in separate files.
+
+Let's examine some code, write it into our `index.html` and run it in the browser. It's yet another version of the **Hello ...** application but please note that we are moving some of the logic from the html markup (using `ng-` directives) to the controller. 
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.6.1/angular.min.js"></script>
+  </head>
+  <body>
+    <div ng-app="demoApp" ng-controller="mainController">
+      <h1>{{'Hello', [firstName, lastName].join(' ')}}!</h1>
+    </div>
+
+    <script>
+      var demoApp = angular.module("demoApp", []);
+
+      demoApp.controller("mainController", function($scope) {
+        $scope.firstName = "Thomas";
+        $scope.lastName = "Ochman";
+      });
+    </script>
+  </body>
+</html>
+```
+
+So, let's go over the code.
+
+1. As we already did in the previous examples, we load AngularJS from CDN.
+2. We tell the `div` tag that it belongs to the `demoApp` and is controlled by `mainController` (Note that in previous examples we left the `ng-app` directive empty and didn't use the `ng-controller` directive at all) 
+3. Within a `<script>` tag, we define the `angular.module` and a `controller` and give them each a name (to be able to reference them). 
+4. In the `mainController` we add two attributes to the `$scope` object (`firstName` & `lastName`).
+5. We add an expression to the `<h1>` tag where we evaluate some JS code, joining the string "Hello" with the two values we assigned to `$scope` and display **"Hello Thomas Ochman"** on the page (we also add an exclamation mark outside of the expression).
+
+### What is `$scope`? 
+Scope (`$scope`) is a JavaScript object that joins the controller with the view and contains the model data. Every controller has an associated `$scope` object.
+
+As an object, `$scope` can have properties and functions.
+
+The scope object is available for both the view and the controller.
+
+In our example we've added 2 properties to `$scope` and we can access them in our view. 
+
+```javascript
+demoApp.controller("mainController", function($scope) {
+        $scope.firstName = "Thomas";
+        $scope.lastName = "Ochman";
+      });
+```
+There's much more to say about `$scope` but for the sake of this introduction we'll stop here. 
```

Comment:
Note that, unlike in Ruby, objects and variables must be passed into controllers in order for them to be available to the application. For instance, above we are passing in `$scope` to the `mainController` - it is not automatically available to use, even though everything is in the same page.